src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6709,7 +6709,14 @@ private StatementSyntax ParseStatementNoDeclaration(bool allowAnyExpression)
                     }
                     else if (isPossibleAwaitUsing())
                     {
-                        return this.ParseUsingStatement(parseAwaitKeywordForAsyncStreams());
+                        if (PeekToken(2).Kind == SyntaxKind.OpenParenToken)
+                        {
+                            return this.ParseUsingStatement(parseAwaitKeywordForAsyncStreams());
+                        }
+                        else
+                        {
+                            return this.ParseLocalDeclarationStatement(parseAwaitKeywordForAsyncStreams());
```

Comment:
>return this.ParseLocalDeclarationStatement(parseAwaitKeywordForAsyncStreams()); [](start = 28, length = 79)

maybe good to add a parsing test for this case

Reply:
We have TestAwaitUsingVarWithVarDeclarationTree, unless I'm misunderstanding you?


---
In reply to: [243707369](https://github.com/dotnet/roslyn/pull/31980#discussion_r243707369) [](ancestors = 243707369)

Reply:
Ah, didn't realize. Thanks